pandas/_libs/testing.pyx

```diff
@@ -194,24 +184,11 @@ cpdef assert_almost_equal(a, b,
             # inf comparison
             return True
 
-        if check_less_precise is True:
-            decimal = 3
-        elif check_less_precise is False:
-            decimal = 5
-        else:
-            decimal = check_less_precise
-
         fa, fb = a, b
 
-        # case for zero
-        if abs(fa) < 1e-5:
-            if not decimal_almost_equal(fa, fb, decimal):
-                assert False, (f'(very low values) expected {fb:.5f} '
-                               f'but got {fa:.5f}, with decimal {decimal}')
-        else:
-            if not decimal_almost_equal(1, fb / fa, decimal):
-                assert False, (f'expected {fb:.5f} but got {fa:.5f}, '
-                               f'with decimal {decimal}')
+        if not math.isclose(fa, fb, rel_tol=rtol, abs_tol=atol):
```

Comment:
is there a perf difference between this and np.allclose (or some other np func)?

Reply:
My understanding is that `np.isclose` / `np.allclose` do other checks so as to deal with arrays, etc. This had been raised in #9457 also. After reading that issue, I got the impression that those checks performed by `assert_*_equals` in pandas are needed, and thus the change should be applied at the level of the actual comparison function.

Also, from the Python docs/[PEP 485](https://www.python.org/dev/peps/pep-0485/), the Python implementation was inspired by numpy's.

Is it meaningful to do a perf test here? Can you provide some advice/guidance on how to do it?

Thanks

Reply:
> Can you provide some advice/guidance on how to do it?

Usually we would use ipython and `%timeit` with one or more example cases that go through the affected code.

Reply:
Will try to find some time during the week to do this. Worst case scenario, I'll do it during the weekend.

Thanks for the suggestion.

Reply:
did you get a measurement here?

Reply:
Yes!

```python
import math
import numpy as np
from pandas import DataFrame

data = []
for a, b in [
    (1e10, 1.00001e10),
    (1e-7, 1e-8),
    (1e10, 1.00001e10),
    (1e-8, 1e-9),
    (1e10, 1.0001e10),
    (1e-8, 1e-9),

]:
    math_isclose = %timeit -q -o math.isclose(a, b)
    np_isclose = %timeit -q -o np.isclose(a, b)
    np_allclose = %timeit -q -o np.allclose(a, b)
    data.append([math_isclose.average, np_isclose.average, np_allclose.average])

DataFrame(data, columns=["math.isclose", "np.isclose", "np.allclose"])
   math.isclose  np.isclose  np.allclose
0  1.271213e-07    0.000032     0.000039
1  1.369607e-07    0.000034     0.000044
2  1.300808e-07    0.000032     0.000039
3  1.255904e-07    0.000033     0.000042
4  1.274097e-07    0.000035     0.000041
5  1.273559e-07    0.000032     0.000041
```

`math.isclose` seems to be consistently 2 orders of magnitude faster than `np.isclose`/`np.allclose`.

I also tried comparing `pandas._libs.testing.assert_almost_equal` with `np.isclose`/`np.allclose`. This was a bit trickier because of the exception being raised:

```python
def ignore_assertion(f, *args, **kwargs):
    try:
        f(*args, **kwargs)
    except AssertionError:
        pass

%timeit np.allclose(df_a, df_b)
# 74.2 µs ± 283 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
%timeit ignore_assertion(assert_almost_equal, df_a.values, df_b.values)
# 1.87 ms ± 39.1 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

# To check the overhead of catching the assertion I also ran the following:
def raise_assert():
    try:
        assert False
    except AssertionError:
        pass

%timeit ignore_assertion(raise_assert)
400 ns ± 3.16 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Does this mean that this PR should be changed to make use of numpy's `np.allclose`? I guess the slower performance is due to the checks performed by `pandas._libs.testing.assert_almost_equal` :thinking: 

WDYT?